src/controllers/userLeaderboard.controller/getUserLeaderboard/index.js

```diff
@@ -0,0 +1,34 @@
+const { validationResult } = require('express-validator');
+const { query: Hasura } = require('../../../utils/hasura');
+const { getUserLeaderboardQuery } = require('./queries/queries');
+const catchAsync = require('../../../utils/catchAsync');
+
+const getUserLeaderboard = catchAsync(async (req, res) => {
+  const sanitizerErrors = validationResult(req);
+  if (!sanitizerErrors.isEmpty()) {
+    return res.status(400).json({
+      success: false,
+      ...sanitizerErrors,
+    });
+  }
+
+  const { cognito_sub } = req.body;
+
+  const response = await Hasura(getUserLeaderboardQuery, {
+    cognitoSub: cognito_sub,
+  });
+
+  const responseData = response.result.data;
+
+  if (!responseData || responseData.user.length === 0) {
```

Comment:
Why are we checking if the user exists or not? Authorizer middleware already checks for this right - so if cognito_sub is present that means the user does exist...?